pkg/controllers/metrics/pod/controller.go

```diff
@@ -0,0 +1,188 @@
+				podGaugeVec.Delete(labels)
+			} else {
+				logging.FromContext(ctx).Debugf("Failed to delete gauge: failed to locate labels")
+			}
+			return reconcile.Result{}, nil
+		}
+		return reconcile.Result{}, err
+	}
+	// Remove the previous gauge after pod labels are updated
+	if labels, ok := c.LabelsMap[req.NamespacedName]; ok {
+		podGaugeVec.Delete(labels)
+	}
+	newlabels, err := c.generateLabels(ctx, pod)
+	if err != nil {
+		logging.FromContext(ctx).Debugf("Failed to generate new labels: %s", err.Error())
+		return reconcile.Result{}, err
+	}
+
+	gauge, err := podGaugeVec.GetMetricWith(newlabels)
+	if err != nil {
+		logging.FromContext(ctx).Debugf("Failed to generate new gauge: %s", err.Error())
+		return reconcile.Result{}, err
+	}
+
+	gauge.Set(float64(1))
+	c.LabelsMap[req.NamespacedName] = newlabels
+
+	return reconcile.Result{}, nil
+}
+
+func (c *Controller) Register(ctx context.Context, m manager.Manager) error {
+	err := controllerruntime.
+		NewControllerManagedBy(m).
+		Named("podmetrics").
+		For(&v1.Pod{}).
+		Complete(c)
+	return err
+}
+
+// generateLabels creates the labels using the current state of the pod
+func (c *Controller) generateLabels(ctx context.Context, pod *v1.Pod) (prometheus.Labels, error) {
+	metricLabels := prometheus.Labels{}
+	metricLabels[podName] = pod.GetName()
+	metricLabels[podNameSpace] = pod.GetNamespace()
+	// Selflink has been deprecated after v.1.20
+	// Manually generate the selflink for the first owner reference
+	// Currently we do not support multiple owner references
+	selflink := ""
+	if len(pod.GetOwnerReferences()) > 0 {
+		ownerreference := pod.GetOwnerReferences()[0]
+		selflink = fmt.Sprintf("/apis/%s/namespaces/%s/%ss/%s", ownerreference.APIVersion, pod.Namespace, strings.ToLower(ownerreference.Kind), ownerreference.Name)
+	}
+	metricLabels[ownerSelfLink] = selflink
+	metricLabels[podHostName] = pod.Spec.NodeName
+	metricLabels[podPhase] = string(pod.Status.Phase)
+	node := &v1.Node{}
+	if err := c.KubeClient.Get(ctx, types.NamespacedName{Name: pod.Spec.NodeName}, node); err != nil {
+		metricLabels[podHostZone] = ""
+		metricLabels[podHostArchitecture] = ""
+		metricLabels[podHostCapacityType] = ""
+		metricLabels[podHostInstanceType] = ""
+		if name, ok := pod.Spec.NodeSelector[v1alpha5.ProvisionerNameLabelKey]; ok {
+			metricLabels[podProvisioner] = name
+		} else {
+			metricLabels[podProvisioner] = ""
+		}
+	} else {
+		metricLabels[podHostZone] = node.Labels[v1.LabelTopologyZone]
+		metricLabels[podHostArchitecture] = node.Labels[v1.LabelArchStable]
+		metricLabels[podHostCapacityType] = node.Labels[v1alpha5.LabelCapacityType]
+		metricLabels[podHostInstanceType] = node.Labels[v1.LabelInstanceTypeStable]
+		metricLabels[podProvisioner] = node.Labels[v1alpha5.ProvisionerNameLabelKey]
+	}
+	// Add pod labels
+	labels, err := json.Marshal(pod.GetLabels())
+	if err != nil {
+		return nil, fmt.Errorf("marshal pod labels: %w", err)
```

Comment:
Same suggestion as above.
```suggestion
		return nil, fmt.Errorf("marshalling pod metric labels: %w", err)
```

Reply:
not including this anymore